packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -0,0 +1,114 @@
+     */
+    constructor(
+        private readonly project: ReaderCollection,
+        private readonly util: MiddlewareUtils,
+        private readonly logger: ToolsLogger
+    ) {}
+
+    /**
+     * @description Handler for reading all fragment files from the workspace
+     * @param _ Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleReadAllFragments = async (_: Request, res: Response, next: NextFunction) => {
+        try {
+            const files = await this.project.byGlob('/**/changes/**/*.fragment.xml');
+
+            if (!files || files.length === 0) {
+                res.status(HttpStatusCodes.OK)
+                    .contentType('application/json')
+                    .send({ fragments: [], message: `No fragments found in the project workspace.` });
+                return;
+            }
+
+            const fragments = files.map((file) => {
+                const fileName = file.getName();
+                return {
+                    fragmentName: fileName
+                };
+            });
+
+            res.status(HttpStatusCodes.OK)
+                .contentType('application/json')
+                .send({
+                    fragments,
+                    message: `${fragments.length} fragments found in the project workspace.`
+                });
+            this.logger.debug(`Read fragments ${JSON.stringify(fragments)}`);
+        } catch (e) {
+            this.logger.error(e.message);
+            res.status(HttpStatusCodes.INTERNAL_SERVER_ERROR).send({ message: e.message });
+            next(e);
+        }
+    };
+
+    /**
+     * Handler for writing a fragment file to the workspace
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleWriteFragment = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const data = req.body as WriteFragmentBody;
+
+            const fragmentName = sanitize(data.fragmentName);
+
+            const sourcePath = this.util.getProject().getSourcePath();
+
+            if (fragmentName) {
+                const fullPath = path.join(sourcePath, FolderNames.Changes, FolderNames.Fragments);
+                const filePath = path.join(fullPath, `${fragmentName}.fragment.xml`);
+
+                if (!fs.existsSync(fullPath)) {
+                    fs.mkdirSync(fullPath);
+                }
+
+                if (fs.existsSync(filePath)) {
+                    res.status(HttpStatusCodes.CONFLICT).send(`Fragment with name "${fragmentName}" already exists`);
+                    return;
+                }
+
+                // Copy the template XML Fragment to the project's workspace
+                const fragmentTemplatePath = path.join(__dirname, '../../templates/rta', TemplateFileName.Fragment);
+                fs.copyFileSync(fragmentTemplatePath, filePath);
+
+                const message = 'XML Fragment created';
```

Comment:
Could you please add logging messages as you nicely did in `handleReadAllFragments`